### ISSUE TITLE
MobileServiceClient.GetPush().RegisterAsync is throwing HTTP 404

### ISSUE THREAD

**Issue by asierpn** [Azure/azure-mobile-apps-net-server#266]
All the requests are being affected (UWP, Xamarin Android and Xamarin iOS). It started yesterday 2018-12-11 02:49:00 +01:00 and it is still happening now.

**Comment by dhei** [comment]
Hey @asierpn,

Sorry for the inconvenience.
This looks like an issue related to App Service backend. We use this GitHub repo to track issues in Azure Mobile Apps .NET Server SDKs. Can you please open a support ticket on Azure portal to get better assistance?

**Comment by guperrot** [comment]
Hi, your server errors should not be related to the SDK updates, please insist on azure support as if it was working before there is no reason it would stop working with the previous SDK. Server errors are most likely reliability issues that Azure support engineers must look at.

As for the client SDK update issues please create an issue on the client repository.

**Comment by asierpn** [comment]
Hi @guperrot,

I'm insisting on Azure support, but it is very coincidental that the issue started on the same day that the new version was published.

The client issue I have is not really an issue, it is a change made in authentication starting in version 4.x which was reported [here](https://github.com/Azure/azure-mobile-apps-net-client/issues/389) and makes the authentication process almost inusable.

**Comment by asierpn** [comment]
I've tested using the last client and server SDKs and the issue still happens...

**Comment by guperrot** [comment]
The issue cannot be caused by a coincidence with sdk releases, publishing a nuget to nuget.org is not having any effect on a service running on Azure.

**Comment by jakeharris4u** [comment]
We also have been getting 404 errors since 12/11 and have been working with Azure support to restore our app to a previous version - even though we have not made any updates to it.  I believe there is something that happened on 12/11 that was not backward compatible in some way that broke all of the routing provided in the common code.  That would explain why others are seeing the same issue at the same time and why our restore process didn't resolve the issue.

@guperrot @dhei Would it be possible to revert the release in an effort to troubleshoot?  Is there anything else that can be done?  Our client relies on this functionality and we don't have any workarounds at this time.  Please help!

**Comment by ConnorMcMahon** [comment]
A hotfix has finished deploying this morning. A site restart should fix this issue.

**Comment by dhei** [comment]
@asierpn, sorry to hear that it's happening again. Please contact Azure support for this issue.

**Comment by asierpn** [comment]
@dhei, yes, I've reopened the Azure ticket, I think that the fix for the authentication issue has broken it again. I've reopened this github ticket also to see if @ConnorMcMahon reads it and takes it into account.

**Comment by ConnorMcMahon** [comment]
I can acknowledge that the regression has popped back up, and I am actively working on a fix that will get deployed shortly.

In the meantime, you can self mitigate by going to https://resources.azure.com. Go to subscriptions > _your subscription_ > Resource Groups > _your resource group_ > providers > Microsoft.Web > sites > _your site_ > config > authsettings. From there, you can edit the subobject "properties" and change "runtimeVersion" to "1.0.3" (save your old setting if at all possible). This will pin you to an old version that we know works. 

Once the fix is finished deploying, then you can restore your old setting for "runtimeVersion". 

Sorry to all for the inconvenience. We are doing everything we can to resolve this as fast as possible.

**Comment by asierpn** [comment]
@ConnorMcMahon I can reach subscriptions > your subscription > Resource Groups > your resource group > providers > Microsoft.Web > sites , but I dont know how to go from sites to "my site", the tree view not expand the sites node.

![image](https://user-images.githubusercontent.com/9281751/52581869-1d81df00-2e2c-11e9-869d-acd704af9598.png)

**Comment by ConnorMcMahon** [comment]
@asierpn that screenshot looks like what I get if I go to subscriptions > your subscription > providers > Microsoft.Web. Are you sure you have selected a resource group?

**Comment by asierpn** [comment]
@ConnorMcMahon yes, I was not selecting the resource group. Now I reach the authsettings node, but there is not a "runtimeVersion" object under "properties". I've tried to add it but this value is not saved.

![image](https://user-images.githubusercontent.com/9281751/52583531-1c52b100-2e30-11e9-931d-0e49f54c92d7.png)

**Comment by ConnorMcMahon** [comment]
Alright, workaround: If you don't have "runtimeVersion" present, then do the following.

1. Set "enabled" to false and POST.
2. Set "enabled" to true and POST.
3. Now "runtimeVersion" should be populated to "1.0.0". Change it to "1.0.3".

**Comment by asierpn** [comment]
@ConnorMcMahon thank you for the workaround, I have been able to change the runtimeVersion object and now is working.

**Comment by ConnorMcMahon** [comment]
@asierpn yes, that would be the right approach.

**Comment by asierpn** [comment]
@ConnorMcMahon I don't know if it is related with the workaround or this issue, but the authentication has stopped working is my sites after this change, and reverting to runtimeVersion 1.0.0 is not helping. The IPrincipal.GetAppServiceIdentityAsync is returning an Invalidoperationexception StatusCode: 404, ReasonPhrase: 'Not Found', Version: 1.1, Content: System.Net.Http.StreamContent

**Comment by ConnorMcMahon** [comment]
@asierpn, You can email me at comcmaho@microsoft.com your app name and I can investigate.

**Comment by asierpn** [comment]
This issue was fixed yesterday by Azure engineers.

**Comment by asierpn** [comment]
Same issue again with GetAppServiceIdentityAsync in development and production. I can not get the claims of the logged user with or without the runtimeVersion setting. Please fix this!

**Comment by asierpn** [comment]
@ConnorMcMahon I've been able to overcome the issue by setting the runtimeVersion to 1.0.3 and getting the claims from the ClaimsPrincipal object instead of from the GetAppServiceIdentityAsync method.

**Comment by ConnorMcMahon** [comment]
@asierpn, can you close this issue, as it does not have to do with the initial topic. Then you could open a separate issue for the `GetAppServiceIdentityAsync()` issue you were having, with as much technical details that you could provide. This would be helpful to improve visibility if anyone else is having the same issue.
